e2e/pageobjects/ide/Editor.ts

```
@@ -179,10 +230,126 @@ export class Editor {
+        const debugBreakpointLocator: By = await this.getDebugBreakpointLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitDisappearanceWithTimeout(debugBreakpointLocator, timeout);
+    }
+
+    async waitBreakpointHint(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointHintLocator: By = await this.getDebugBreakpointHintLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitVisibility(debugBreakpointHintLocator, timeout);
+    }
+
+    async waitBreakpointHintDisappearance(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointHintLocator: By = await this.getDebugBreakpointHintLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitDisappearanceWithTimeout(debugBreakpointHintLocator, timeout);
+    }
+
+    async activateBreakpoint(tabTitle: string,
+        lineNumber: number,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+
+        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
+        const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+
+        for (let i = 0; i < attempts; i++) {
+            const elementLocator: By = await this.getLineNumberBlockLocator(tabTitle, lineNumber);
+            const element: WebElement = await this.driverHelper.waitVisibility(elementLocator, timeout);
+
+            try {
+                await this.driverHelper.getAction().mouseMove(element, { x: 5, y: 5 }).perform();
+                await this.waitBreakpointHint(tabTitle, lineNumber);
+                await this.driverHelper.getAction().click().perform();
+                await this.waitBreakpoint(tabTitle, lineNumber);
+                return;
+            } catch (err) {
+                // ignore errors and wait
+                await this.driverHelper.wait(polling);
```

Do we know what errors can happen here? Shouldn't we at least log them?

---

In this case, we should ignore all exceptions

---

Yeah.. I can see that from the code, that we are ignoring all the errors... My question was "Why?" 
I'm not saying this "silent swallowing" of errors *have to be* wrong. But usually it's a bad code smell and I think it needs more explanation.
Furthermore, there is quite a big chunk (relatively) of code in the try-catch block and I can imagine quite lot of possible errors could happen - having them logged could help in possible debugging.

---

Because actions inside of try/catch are very sensitive, and I tried to process most common exceptions and final was that the "WebDriverError" may be happening and it does not make sense to process other exceptions because it is a parent of all driver exceptions. Anyway, if something went wrong exception with full explanation will be thrown in the further code.